pkg/activator/handler/handler_test.go

```diff
@@ -219,12 +248,21 @@ func TestActivationHandler(t *testing.T) {
 		attempts:        "hi there",
 		endpointsGetter: brokenEndpointGetter,
 		reporterCalls:   nil,
-	},
-	}
+	}}
 
 	for _, e := range examples {
 		t.Run(e.label, func(t *testing.T) {
 			rt := util.RoundTripperFunc(func(r *http.Request) (*http.Response, error) {
+				if r.Header.Get(network.ProbeHeaderName) != "" {
+					if e.probeErr != nil {
+						return nil, e.probeErr
+					}
+					fake := httptest.NewRecorder()
+					fake.WriteHeader(http.StatusOK)
```

Comment:
Do we want a test case that will do, say, http.StatusNotFound or InternalServerError?

Reply:
I added one for 500s